client/python/tests/integration/test_no_auth.py

```diff
@@ -1,113 +1,162 @@
 
-def test_submit_job():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
-    )
-    # Jobs can must be finished before deleting queue
-    no_auth_client.cancel_jobs(queue=queue_name, job_set_id=job_set_name)
-    no_auth_client.delete_queue(name=queue_name)
-    time.sleep(1)
+    if server_ssl == "true":
+        channel_credentials = grpc.ssl_channel_credentials()
+        return ArmadaClient(
+            channel=grpc.secure_channel(
+                f"{server_name}:{server_port}", channel_credentials
+            )
+        )
 
+    else:
+        return ArmadaClient(
+            channel=grpc.insecure_channel(f"{server_name}:{server_port}")
+        )
 
-def test_submit_job_and_cancel_by_id():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    jobs = no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
-    )
-    # Jobs can must be finished before deleting queue
-    no_auth_client.cancel_jobs(job_id=jobs.job_response_items[0].job_id)
-    no_auth_client.delete_queue(name=queue_name)
 
+no_auth_client = client()
 
-def test_submit_job_and_cancel_by_queue_job_id():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
+queue_name = f"queue-{uuid.uuid1()}"
+
+
+def submit_sleep_job():
+    pod = core_v1.PodSpec(
+        containers=[
+            core_v1.Container(
+                name="sleep",
+                image="alpine:latest",
+                args=["sleep", "30s"],
+                resources=core_v1.ResourceRequirements(
+                    requests={
+                        "cpu": api_resource.Quantity(string="0.2"),
+                        "memory": api_resource.Quantity(string="64Mi"),
+                    },
+                    limits={
+                        "cpu": api_resource.Quantity(string="0.2"),
+                        "memory": api_resource.Quantity(string="64Mi"),
+                    },
+                ),
+            )
+        ],
     )
-    # Jobs can must be finished before deleting queue
-    no_auth_client.cancel_jobs(queue=queue_name, job_set_id=job_set_name)
-    no_auth_client.delete_queue(name=queue_name)
 
+    return [no_auth_client.create_job_request_item(priority=0, pod_spec=pod)]
 
-def test_get_queue():
-    queue_name = f"queue-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
 
+def wait_for(queue=None, job_set_id=None):
```

Comment:
I think you should refactor this into two functions.  

A good approach to follow is SRP (Single Responsibility Principle).  A function should do one thing.

Reply:
Okay that's fine by me :)

Reply:
Once you address this, I think it looks good to me!

Reply:
On a slack message, we decided to leave this as is.

Reply:
Can we at least guard against both values being set, if that's an invalid usage?

Reply:
I don't think that is an invalid usage, I think its fine for both to be set.

Reply:
After re-writing this code with fixtures, I have re-written it to be dependent on just the job-set=id being present, as the queue var needs to be present in either case.